src/appleseed/renderer/kernel/intersection/intersector.cpp

```diff
@@ -224,13 +224,27 @@ namespace
         const ShadingPoint&         shading_point,
         const ShadingPoint*         parent_shading_point)
     {
+        static const size_t MaxWarningsPerThread = 100;
+        static size_t warnings_count = 0;

```

Comment:
If you put "count" after something, that something must be singular: `warning_count`.